src/v2/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkActions.tsx

```diff
@@ -377,23 +386,50 @@ export const UtilButton: React.ForwardRefExoticComponent<
           <ActionIcon {...rest} fill="currentColor" />
         </Flex>
 
-        {label && (
-          <Text variant="xs" lineHeight={1}>
-            {label}
-          </Text>
-        )}
+        <UtilButtonInnerText
+          label={label}
+          longestLabel={longestLabel}
+          variant="xs"
+          lineHeight={1}
+        />
       </Component>
     )
   }
 )
 
+const UtilButtonInnerText: React.FC<UtilButtonInnerTextProps> = ({
+  label,
+  longestLabel,
+  ...rest
+}) => {
+  if (!label) {
+    return null
+  }
+
+  if (longestLabel) {
+    return (
+      <Box position="relative">
+        <VisibleText className="visibleText" {...rest}>
+          {label}
+        </VisibleText>
+        <HiddenText role="presentation" {...rest}>
+          {longestLabel}
+        </HiddenText>
+      </Box>
+    )
+  }
+
+  return <Text {...rest}>{label}</Text>
+}
+
 const utilButtonMixin = css`
   display: flex;
   align-items: center;
   justify-content: center;
   color: ${themeGet("colors.black100")};
 
-  &:hover {
+  &:hover,
+  &:hover .visibleText {
```

Comment:
Here these changes are necessary in order for `text-decoration: underline` to be applied to `VisibleText` which has `position: absolute`. Otherwise, when hover over the element, the text will not have underline ([here](https://stackoverflow.com/a/39491131) you can find more context)

https://user-images.githubusercontent.com/3513494/171870091-cd12b4da-0b48-473b-a1c1-1cc7251021cf.mp4

Reply:
You can target it with `${VisibleText}` if you want, instead of using classes.

Reply:
oh, good advice. thank you!